### PR TITLE
misc dungeon profit changes

### PR DESCRIPTION
**Pull request by efefury**
- changed the location of buttons when in dungeon profit overlay
- added a warning if derpy is mayor
image

### CODE REVIEW

File: src/main/java/io/github/moulberry/notenoughupdates/util/SBInfo.java [hunk 1]
```diff
@@ -294,6 +299,14 @@ public void tick() {
 			lastLocRaw = System.currentTimeMillis();
 			NotEnoughUpdates.INSTANCE.sendChatMessage("/locraw");
 		}
+		try {
+		if (currentTime - lastMayorUpdate > 300 * 1000) {
+			mayorJson = Utils.getCurrentMayor();
+			lastMayorUpdate = currentTime;
+		}
+			} catch (ExecutionException | InterruptedException e) {
+				throw new RuntimeException(e);
```
Comment (Lulonaut) [inline src/main/java/io/github/moulberry/notenoughupdates/util/SBInfo.java:308]: Crashing the users game when something goes wrong with the request does not seem like a good idea

File: src/main/java/io/github/moulberry/notenoughupdates/util/Utils.java [hunk 2]
```diff
@@ -1966,6 +1969,16 @@ public static void showOutdatedRepoNotification() {
 	 * 6 = SPECIAL
 	 * 7 = VERY SPECIAL
 	 */
+
+	public static JsonObject getCurrentMayor() throws ExecutionException, InterruptedException {
```
Comment (Lulonaut) [inline src/main/java/io/github/moulberry/notenoughupdates/util/Utils.java:1973]: I would move this to SBInfo.java, does not seem like a Utils thing

**Review by Lulonaut — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/moulberry/notenoughupdates/util/SBInfo.java [hunk 3]
```diff
@@ -418,4 +430,17 @@ public void tick() {
...
 		}
 	}
+
+	public static JsonObject getCurrentMayor() throws ExecutionException, InterruptedException {
+		CompletableFuture<JsonObject> hypixelApiAsync = NotEnoughUpdates.INSTANCE.manager.hypixelApi.getHypixelApiAsync(
+			NotEnoughUpdates.INSTANCE.config.apiData.apiKey,
+			"resources/skyblock/election",
+			new HashMap<>()
+		);
+		return hypixelApiAsync.get();
```
Comment (romangraef) [inline src/main/java/io/github/moulberry/notenoughupdates/util/SBInfo.java:440]: Can we not use blocking methods in a `tick()` method?

**Review by romangraef — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/moulberry/notenoughupdates/util/SBInfo.java [hunk 4]
```diff
@@ -431,15 +431,15 @@ public void tick() {
 		}
 	}
 
-	public static JsonObject getCurrentMayor() throws ExecutionException, InterruptedException {
-		CompletableFuture<JsonObject> hypixelApiAsync = NotEnoughUpdates.INSTANCE.manager.hypixelApi.getHypixelApiAsync(
+	public void getCurrentMayor() throws ExecutionException, InterruptedException {
```
Comment (romangraef) [inline src/main/java/io/github/moulberry/notenoughupdates/util/SBInfo.java:434]: Do we still throw those exceptions? Also i feel like we could give this function a better name now that it doesnt return anything anymore

**Review by romangraef — APPROVED** [review]

**Review by Lulonaut — APPROVED** [review]

**Review by NopoTheGamer — APPROVED** [review]

**Pull request merged**